cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -103,6 +103,14 @@ func RemoveStackedEtcdMemberFromCluster(client clientset.Interface, cfg *kubeadm
 		return err
 	}
 
+	// If this is the only remaining stacked etcd member in the cluster, calling RemoveMember()
+	// is not needed.
+	etcdClientAddress := etcdutil.GetClientURL(&cfg.LocalAPIEndpoint)
+	if len(etcdClient.Endpoints) == 1 && etcdClient.Endpoints[0] == etcdClientAddress {
```

Comment:
we had a discussion in the kubeadm meeting about this:
- etcdClient.Endpoints is not a good way to check this because the user might have configured multiple endpoints, instead we should add a function there that returns `[]Members`:
https://github.com/kubernetes/kubernetes/blob/b339ed0e75c5815d49c5c4d118d4b9858b4babc4/cmd/kubeadm/app/util/etcd/etcd.go
- then if the list of members contains only a single member we can for-loop it's enpoints and see if one of them matches the `etcdClientAddress`

@tnqn 
please let me know if you have further questions.

/cc @fabriziopandini

Reply:
@neolit123 thanks for the suggestion, updated, PTAL.